lib/logstash/inputs/logstash.rb

```diff
@@ -102,6 +102,7 @@ def inner_http_input_options
       if @username
         http_options['user'] = @username
         http_options['password'] = @password || report_invalid_config!('`password` is REQUIRED when `username` is provided')
+        report_invalid_config!("Empty `username` or `password` is not allowed") if @username.empty? || @password.value.empty?
```

Comment:
Where does this requirement come from?

Reply:
As we took offline, changed back to support empty password.